testing/trino-tests/src/test/java/io/trino/security/TestAccessControl.java

```diff
@@ -617,4 +617,32 @@ public void testSetViewAuthorizationWithSecurityInvoker()
     {
         assertQuerySucceeds("ALTER VIEW mock.default.test_view_invoker SET AUTHORIZATION some_other_user");
     }
+
+    @Test
+    public void testUseStatementAccessControl()
+    {
+        executeExclusively(() -> {
+            Session session = testSessionBuilder()
+                    .setCatalog(Optional.empty())
+                    .setSchema(Optional.empty())
+                    .build();
+            getQueryRunner().execute(session, "USE tpch.tiny");
+
+            assertThatThrownBy(() -> getQueryRunner().execute("USE not_exists_catalog.tiny"))
+                    .hasMessageMatching("Catalog does not exist: not_exists_catalog");
```

Comment:
Do we need a test where we deny access to a not existing catalog/schema and see that the message is "catalog/schema does not exist" instead of "access denied"? Is this part of the contract or just a consistency thing? @kokosing

Reply:
https://github.com/trinodb/trino/pull/14209#discussion_r975759257